conftest.py

```diff
@@ -191,35 +196,57 @@ def pytest_runtest_call(item):
     # Get test-specific params from potentially locally defined fixtures
     eips = item.funcargs["eips"]
     fork = item.funcargs["fork"]
-    if "state_test" in item.funcargs:
-        test_spec = item.funcargs["state_test"].spec
-    elif "blockchain_test" in item.funcargs:
-        test_spec = item.funcargs["blockchain_test"].spec
-    else:
-        # TODO: Raise custom exception instead of fail to trigger a
-        # pytest error
-        pytest.fail(
-            "Invalid test function signature: Test did not use either "
-            " of the 'state_test' or 'blockchain_test' fixtures."
-        )
 
     if not t8n.is_fork_supported(fork):
         pytest.skip(f"Fork '{fork}' not supported by t8n, skipped")
     if fork == ArrowGlacier:
         pytest.skip(f"Fork '{fork}' not supported by hive, skipped")
 
-    fixture_name = convert_pytest_case_name_to_fixture_name(item)
-    fixture_output = fill_test(
-        fixture_name,
-        t8n,
-        b11r,
-        test_spec,
-        fork,
-        engine,
-        reference_spec,
-        eips=eips,
-    )
-    write_fixture_file(item, fixture_output)
+    if "state_test" in item.funcargs:
+
+        class StateTestWrapper(StateTest):
+            def __init__(self, *args, **kwargs):
+                super(StateTestWrapper, self).__init__(*args, **kwargs)
+                fixture_name = convert_pytest_case_name_to_fixture_name(item)
+                fixture_output = fill_test(
+                    fixture_name,
+                    t8n,
+                    b11r,
+                    self,
+                    fork,
+                    engine,
+                    reference_spec,
+                    eips=eips,
+                )
+                write_fixture_file(item, fixture_output)
+
+        item.funcargs["state_test"] = StateTestWrapper
+
+    if "blockchain_test" in item.funcargs:
+
+        class BlockchainTestWrapper(StateTest):
```

Comment:
Minor typo, the wrapper should derive from `BlockchainTest` not `StateTest` here.

Reply:
Will merge to the main pytest branch and fix there!

Reply:
This is fixed via https://github.com/marioevz/execution-spec-tests/pull/4.